spec/build-npm_spec.sh

```diff
@@ -0,0 +1,217 @@
+  else
+    echo "jq $*"
+  fi
+End
+
+Mock jf
+  if [[ "$*" == "--version" ]]; then
+    echo "jf version 2.77.0"
+  elif [[ "$*" == "rt build-publish test-project 42" ]]; then
+    echo '{"buildInfoUiUrl": "https://repox.jfrog.io/ui/builds/test-project/42/123456/published"}'
+  else
+    echo "jf $*"
+  fi
+End
+
+Mock npm
+  if [[ "$*" == "--version" ]]; then
+    echo "10.2.4"
+  elif [[ "$*" == "ci" ]]; then
+    echo "npm ci completed"
+  elif [[ "$*" == "test" ]]; then
+    echo "npm test completed"
+  elif [[ "$*" == "run build" ]]; then
+    echo "npm run build completed"
+  elif [[ "$*" =~ ^version ]]; then
+    echo "npm version completed"
+  else
+    echo "npm $*"
+  fi
+End
+
+Mock npx
+  if [[ "$*" =~ sonarqube-scanner ]]; then
+    echo "SonarQube scanner completed"
+  else
+    echo "npx $*"
+  fi
+End
+
+Mock git
+  echo "git $*"
+End
+
+# Set up environment variables
+export GITHUB_REPOSITORY="my-org/test-project"
+export GITHUB_REF_NAME="main"
+export GITHUB_EVENT_NAME="push"
+export BUILD_NUMBER="42"
+export GITHUB_RUN_ID="12345"
+export GITHUB_SHA="abc123"
+export GITHUB_OUTPUT=/dev/null
+export ARTIFACTORY_URL="https://repox.jfrog.io/artifactory"
+export ARTIFACTORY_DEPLOY_REPO="test-repo"
+export ARTIFACTORY_DEPLOY_ACCESS_TOKEN="test-token"
+export SONAR_HOST_URL="https://sonar.example.com"
+export SONAR_TOKEN="sonar-token"
+export DEPLOY_PULL_REQUEST="false"
+export SKIP_TESTS="false"
+GITHUB_EVENT_PATH=$(mktemp)
+export GITHUB_EVENT_PATH
+echo '{}' > "$GITHUB_EVENT_PATH"
+
+# Create mock package.json
+echo '{"version": "1.2.3-SNAPSHOT", "name": "test-project"}' > package.json
+
+Describe 'build-npm/build.sh'
+  It 'does not run build_npm() if the script is sourced'
+    When run source build-npm/build.sh
+    The status should be success
+    The output should equal ""
+  End
+
+  It 'runs main() when executed directly'
+    When run script build-npm/build.sh
+    The status should be success
+    The output should include "=== NPM Build, Deploy, and Analyze ==="
+    The output should include "Branch: main"
+    The output should include "Pull Request: false"
+    The stderr should include "WARN: Version '1.2.3-42' does not match the expected format"
+  End
+
+  Describe 'Git utility functions'
+    BeforeEach 'source build-npm/build.sh'
+
+    It 'detects main branch correctly'
+      export GITHUB_REF_NAME="main"
+      When call is_main_branch
+      The status should be success
+    End
+
+    It 'detects master branch correctly'
+      export GITHUB_REF_NAME="master"
+      When call is_main_branch
+      The status should be success
+    End
+
+    It 'detects maintenance branch correctly'
+      export GITHUB_REF_NAME="branch-1.2"
+      When call is_maintenance_branch
+      The status should be success
+    End
+
+    It 'detects pull request correctly'
+      export PULL_REQUEST="123"
+      When call is_pull_request
+      The status should be success
+    End
+  End
```

Comment:
Is this needed for the coverage, or was it useful for development/debug?

We likely don't need to test that `GITHUB_REF_NAME="branch-1.2"` will match `[[ "${GITHUB_REF_NAME}" == "branch-"* ]]`
This is not absurd per se, but it is duplicated among the tests, covered by the other higher level tests, and a bit too much obvious.